### ISSUE TITLE
RSA keys used for travis encrypt are too small

### ISSUE THREAD

**Issue by alex** [travis-ci/travis-ci#3085]
Right now the keys (for every repo I've looked at) are 1024 bits. This is generally considered too small for secure usage -- it's about 73 bits of security (see http://www.ecrypt.eu.org/documents/D.SPA.20.pdf page 41). It would be better to upgrade these keys to 2048 bit or larger, which provides a greater security margin.

**Comment by dstufft** [comment]
Alternatively, if you still want to have smaller key sizes you can use something like Curve25519 to encrypt which will give better security with smaller keys. https://github.com/cryptosphere/rbnacl can be used with the public key encryption to do that.

**Comment by alex** [comment]
Curve25519 and the Box recipe require the encryptor to have their own
keypair, and travis to track the public key of the person who encrypted the
contents. That isn't part of the existing model.

**Comment by alex** [comment]
Pinging on this.

**Comment by mmlb** [comment]
Also is there a way to force regeneration of keys? I've got one repo with a 1024 bit pubkey and a newer one with 4096 bit pubkey.

**Comment by stephanmg** [comment]
Hey,
I got my travis key by:
travis pubkey

However if I add it to my SSH server's authorized keys file, it still will ask for a password, when
I want to scp something from my SSH server into the Travis builder folder.
Is this to be expected?

Best Stephan

**Comment by alex** [comment]
Pinging on this again.

I intend to write a blog post describing this problem if it is not resolved soon.

**Comment by BanzaiMan** [comment]
@alex Sorry for the delay. We have an internal ticket tracking this issue.

**Comment by samrocketman** [comment]
Indeed, I've stopped storing secrets in Travis format because of this.  Instead, I've opted for other means.

**Comment by alex** [comment]
Just to complete the thought on nacl's Box abstraction, you can work around the need for a public key by generating an adhoc key pair on the client, doing the Box operation, appending the public key, and then throwing away the private key.

**Comment by alex** [comment]
This issue still hasn't been resolved to the best of my knowledge.

**Comment by samrocketman** [comment]
Pretty sure Travis CI still uses 1024 bit RSA keys.

**Comment by glyph** [comment]
It looks like newer repositories get 4096-bit keys, but there's no way to rotate old keys, so old repositories still have 1024-bit ones.

**Comment by graingert** [comment]
Maybe deleting the Travis repo, and re-adding it?
